digdag-core/src/main/java/io/digdag/core/workflow/WorkflowExecutor.java

```diff
@@ -852,20 +848,42 @@ private void enqueueTask(final TaskQueueDispatcher dispatcher, final long taskId
         }).or(false);
     }
 
+    private static String encodeUniqueQueuedTaskName(StoredTask task)
+    {
+        Instant updatedAt = task.getUpdatedAt();
+        return Long.toString(task.getId()) + ".t" + String.format(ENGLISH,
+                "%x%08x", updatedAt.getEpochSecond(), updatedAt.getNano());
```

Comment:
`updated_at` (generated using `now()` pg function) is not guaranteed to be unique and `now()` precision is not guaranteed to be high.

https://www.postgresql.org/message-id/9699.1262011789@sss.pgh.pa.us

This might especially be true if pg is running in a VM etc.

Perhaps we could use a counter instead?